src/isValidCommitMesage.ts

```diff
@@ -1,42 +1,12 @@
-const DEFAULT_COMMIT_TYPES = [
-    "feat",
-    "fix",
-    "docs",
-    "style",
-    "refactor",
-    "test",
-    "build",
-    "perf",
-    "ci",
-    "chore",
-    "revert",
-    "merge",
-    "wip",
-];
-
-const isValidCommitMessage = (message, availableTypes = DEFAULT_COMMIT_TYPES): boolean => {
+const isValidCommitMessage = (message): boolean => {
     // Exceptions.
     // This is a message that's auto-generated by git. Can't do much about it unfortunately. Let's allow it.
     if (message.startsWith("Merge ") || message.startsWith("Revert ")) {
         return true;
     }
 
-    // Commit message doesn't fall into the exceptions group. Let's do the validation.
-    let [possiblyValidCommitType] = message.split(":");
-    possiblyValidCommitType = possiblyValidCommitType.toLowerCase();
-
-    // Let's remove scope if present.
-    if (possiblyValidCommitType.match(/\(\S*?\)/)) {
-        possiblyValidCommitType = possiblyValidCommitType.replace(/\(\S*?\)/, "");
-    }
-
-    possiblyValidCommitType = possiblyValidCommitType
-        .replace(/\s/g, "") // Remove all whitespace
-        .replace(/\!/g, "") // Remove bang for notify breaking change
-        .replace(/()/g, "") // Remove all whitespace
-        .replace(/[^a-z]/g, ""); // Only leave [a-z] characters.
-
-    return availableTypes.includes(possiblyValidCommitType);
+    let pattern = /^(build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test){1}(\([\w\-\.]+\))?(!)?: \S+([\s\S]*)/
```

Comment:
I'd rather the `DEFAULT_COMMIT_TYPES` be as it is and use 
```
commitPrefixes = DEFAULT_COMMIT_TYPES.join("|");
```
to build the pattern. This makes it easier to extend the list of valid commit prefixes and improves readability.